spec/integration/type/file_spec.rb

```diff
@@ -1528,4 +1528,161 @@ def pretty_transaction_error(transaction)
 
     raise "Got #{status_failures.length} failure(s) while applying: #{status_fail_msg}"
   end
+
+  describe "copying a file that is a link to a file" do
+    let(:target) { tmpfile('target') }
+    let(:link) { tmpfile('link') }
+    let(:copy) { tmpfile('copy') }
+    it "should copy the target of the link if :links => follow" do
+      catalog.add_resource described_class.new(
+        :name => target,
+        :ensure => "present",
+        :content => "Jenny I got your number / I need to make you mine")
+      catalog.add_resource described_class.new(
+        :name => link,
+        :ensure => "link",
+        :target => target)
+      catalog.add_resource described_class.new(
+        :name => copy,
+        :ensure => "present",
+        :source => link,
+        :links => "follow")
+      catalog.apply
+      expect(File.ftype(target)).to eq(File.ftype(copy))
```

Comment:
Yep, `File.ftype` on Windows works for files and directories, but not symlinks on systems that support them:

```
C:\work\puppet> bundle exec puppet resource file C:\work\puppet\link ensure=link target=c:\work\puppet\target
Notice: /File[C:\work\puppet\link]/ensure: created
file { 'C:\work\puppet\link':
  ensure => 'link',
  target => 'c:\work\puppet\target',
}
C:\work\puppet> bundle exec irb
irb(main):001:0> File.ftype('c:\work\puppet\link')
=> "file"
irb(main):002:0> require 'puppet'
=> true
irb(main):003:0> Puppet::FileSystem.symlink?('c:\work\puppet\link')
=> true
```

And actually a better way of doing this would be to use:

```
irb(main):008:0> Puppet::FileSystem.lstat('c:\work\puppet\link').ftype
=> "link"
```